### PR TITLE
Feature 7 offline surveys

### LINKED ISSUE #7: Offlining surveys
As a user I'd like the device to be able to offline surveys

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #7

### CODE REVIEW

File: lib/mqtt/listeners/abstract_listener.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'package:meta/meta.dart';
+import 'package:oss_surveys_customer/mqtt/mqtt_client.dart';
```
Comment (EVP1989) [inline lib/mqtt/listeners/abstract_listener.dart]:
```suggestion
import "package:meta/meta.dart";
import "package:oss_surveys_customer/mqtt/mqtt_client.dart";
```

File: lib/mqtt/model/status_message.dart [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import 'package:oss_surveys_customer/mqtt/model/abstract_message.dart';
```
Comment (EVP1989) [inline lib/mqtt/model/status_message.dart]:
```suggestion
import "package:oss_surveys_customer/mqtt/model/abstract_message.dart";
```

File: lib/mqtt/model/survey_message.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import 'package:oss_surveys_customer/mqtt/model/abstract_message.dart';
```
Comment (EVP1989) [inline lib/mqtt/model/survey_message.dart]:
```suggestion
import "package:oss_surveys_customer/mqtt/model/abstract_message.dart";
```

File: lib/mqtt/mqtt_client.dart [hunk 4]
```diff
@@ -54,11 +56,15 @@ class MqttClient {
       if (c.isEmpty) {
         return;
       }
-      final MqttPublishMessage message = c[0].payload as MqttPublishMessage;
-      final payload =
-      MqttPublishPayload.bytesToStringAsString(message.payload.message);
 
-      logger.info("Received message:$payload from topic: ${c[0].topic}>");
+      if (listeners.containsKey(c[0].topic)) {
```
Comment (EVP1989) [inline lib/mqtt/mqtt_client.dart]: Consider using better describin name than "c"

**Review by EVP1989 — APPROVED** [review]
Approved with some comments/suggetions

File: lib/mqtt/listeners/abstract_listener.dart (hunk 1 above)
Comment (villejuutila) [reply lib/mqtt/listeners/abstract_listener.dart]: Resolved

File: lib/mqtt/model/survey_message.dart (hunk 3 above)
Comment (villejuutila) [reply lib/mqtt/model/survey_message.dart]: Resolved

File: lib/mqtt/mqtt_client.dart (hunk 4 above)
Comment (villejuutila) [reply lib/mqtt/mqtt_client.dart]: Resolved

File: lib/mqtt/model/status_message.dart (hunk 2 above)
Comment (villejuutila) [reply lib/mqtt/model/status_message.dart]: Resolved

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**